packages/frontend-api/src/Model/Mutation/Customer/User/CustomerUserMutation.php

```diff
@@ -0,0 +1,144 @@
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Customer\User\FrontendCustomerUserProvider $frontendCustomerUserProvider
+     * @param \Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface $userPasswordEncoder
+     * @param \Shopsys\FrameworkBundle\Model\Customer\User\CustomerUserPasswordFacade $customerUserPasswordFacade
+     * @param \Shopsys\FrameworkBundle\Model\Customer\User\CustomerUserRefreshTokenChainFacade $customerUserRefreshTokenChainFacade
+     * @param \Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface $tokenStorage
+     * @param \Shopsys\FrontendApiBundle\Model\Customer\User\CustomerUserUpdateDataFactory $customerUserUpdateDataFactory
+     * @param \Shopsys\FrameworkBundle\Model\Customer\User\CustomerUserFacade $customerUserFacade
+     */
+    public function __construct(
+        FrontendCustomerUserProvider $frontendCustomerUserProvider,
+        UserPasswordEncoderInterface $userPasswordEncoder,
+        CustomerUserPasswordFacade $customerUserPasswordFacade,
+        CustomerUserRefreshTokenChainFacade $customerUserRefreshTokenChainFacade,
+        TokenStorageInterface $tokenStorage,
+        CustomerUserUpdateDataFactory $customerUserUpdateDataFactory,
+        CustomerUserFacade $customerUserFacade
+    ) {
+        $this->frontendCustomerUserProvider = $frontendCustomerUserProvider;
+        $this->userPasswordEncoder = $userPasswordEncoder;
+        $this->customerUserPasswordFacade = $customerUserPasswordFacade;
+        $this->customerUserRefreshTokenChainFacade = $customerUserRefreshTokenChainFacade;
+        $this->tokenStorage = $tokenStorage;
+        $this->customerUserUpdateDataFactory = $customerUserUpdateDataFactory;
+        $this->customerUserFacade = $customerUserFacade;
+    }
+
+    /**
+     * @param \Overblog\GraphQLBundle\Definition\Argument $argument
+     */
+    public function changePassword(Argument $argument)
+    {
+        $input = $argument['input'];
+
+        try {
+            $customerUser = $this->frontendCustomerUserProvider->loadUserByUsername($input['email']);
+        } catch (UsernameNotFoundException $e) {
+            throw new UserError('User does not exists or provided password is not valid.');
+        }
+
+        if (!$this->userPasswordEncoder->isPasswordValid($customerUser, $input['oldPassword'])) {
+            throw new UserError('User does not exists or provided password is not valid.');
+        }
+
+        $this->customerUserPasswordFacade->changePassword($customerUser, $input['newPassword']);
+
+        return $customerUser;
+    }
+
+    /**
+     * @param \Overblog\GraphQLBundle\Definition\Argument $argument
+     * @return \Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser
+     */
+    public function changePersonalData(Argument $argument): CustomerUser
+    {
+        $token = $this->tokenStorage->getToken();
+
+        if ($token === null) {
+            throw new UserError('Unlogged user');
```

Comment:
I used same user error as in LogoutMutation to be consistent.

Reply:
Maybe you could change sides. I do not really thing that unprovided token and wrong token should end both with error message `Unlogged user` - again it is getting user experience worse.
When it is also in `LogoutMutation` it may be fixed there too.

Reply:
Ok I changed it to `Token is not valid` in all places

Reply:
OK, but not really OK.